test/test_utils_test.go

```diff
@@ -207,3 +217,117 @@ func (s *keysPropagator) ExtractToWorkflow(ctx workflow.Context, reader workflow
 	}
 	return ctx, nil
 }
+
+func (ts *ConfigAndClientSuiteBase) InitConfigAndClient() error {
+	ts.config = NewConfig()
+	var err error
+	err = WaitForTCP(time.Minute, ts.config.ServiceAddr)
+	if err != nil {
+		return err
+	}
+	ts.client, err = client.Dial(client.Options{
```

Comment:
This is causing a goroutine leak w/ an unclosed client in the integration test which I think is causing your test timeout. You are calling this in `SetupSuite` which creates this client in the field at the suite level, but that never happened before. The client is meant to be created _per test_ which it is in `SetupTest` and closed in `TearDownTest`.

Don't set this field in suite-level init, only in test-level init. And always make sure created clients are closed.

Reply:
Thanks